src/pyEQL/engines.py

```diff
@@ -665,23 +665,50 @@ def equilibrate(self, solution: "solution.Solution") -> None:
         # store the original solvent mass
         orig_solvent_moles = solution.components[solution.solvent]
 
+        # store the original solvent elements and components with that element
+        # that we may need to add back later.
+        orig_el_dict = solution.get_el_amt_dict(nested=True)
+        orig_components_by_element = solution.get_components_by_element(nested=True)
+
+        solution.components = FormulaDict({})
         # use the output from PHREEQC to update the Solution composition
         # the .species_moles attribute should return MOLES (not moles per ___)
         for s, mol in self.ppsol.species_moles.items():
             solution.components[s] = mol
 
-        # make sure all species are accounted for
-        assert set(self._stored_comp.keys()) - set(solution.components.keys()) == set()
-
         # log a message if any components were not touched by PHREEQC
         # if that was the case, re-adjust the charge balance to account for those species (since PHREEQC did not)
         missing_species = set(self._stored_comp.keys()) - {standardize_formula(s) for s in self.ppsol.species}
         if len(missing_species) > 0:
             logger.warning(
-                f"After equilibration, the amounts of species {missing_species} were not modified "
+                f"After equilibration, the amounts of species {sorted(missing_species)} were not modified "
                 "by PHREEQC. These species are likely absent from its database."
             )
 
+        # tolerance (in moles) for detecting cases where an element amount
+        # is no longer balanced because of species that are not recognized
+        # by PHREEQC.
+        _atol = 1e-16
+
+        for el in orig_el_dict:
+            orig_el_amount = sum([orig_el_dict[el][k] for k in orig_el_dict[el]])
+            new_el_amount = solution.get_total_amount(el, units="mol").magnitude
```

Comment:
Here, would it be more efficient to call `get_el_amt_dict` (outside the loop) and then look up items from it while iterating? B/c I think using `get_total_amount` will call `get_el_amt_dict ` every time.